src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1284,10 +1284,24 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
 
   def forceCountRows(): Long = rvd.count()
 
-  def deduplicate(): MatrixTable =
+  def distinctByRow(): MatrixTable =
     copy2(rvd = rvd.boundary.mapPartitionsPreservesPartitioning(rvd.typ,
       SortedDistinctRowIterator.transformer(rvd.typ)))
 
+  def distinctByCol(): MatrixTable = {
+    val colKeys = dropRows().colKeys
```

Comment:
Why do you need to dropRows() first?

Reply:
using `colValues` forces the evaluation of the MatrixValue. This will make that evaluation potentially much cheaper.

Reply:
that makes sense.